File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return
        minLength = math.inf
        for str in strs:
            minLength = min(minLength, len(str))
        
        res = 0
        for i in range(0, minLength):
            letter = strs[0][i:i+1]
            same = True
            for str in strs:
                if str[i:i+1] != letter:
                    same = False
                    break
            if not same:
                return strs[0][0:i]
        return strs[0][0:minLength]

File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
const longestCommonPrefix = function(strs) {
    let minLength = Infinity;
    for (let str of strs) {
        minLength = Math.min(minLength, str.length);
    }
    for (let i = 0; i < minLength; i++) {
        let letter = strs[0].charAt(i);
        let same = true;
        for (let j = 1; j < strs.length; j++) {
            if (letter !== strs[j].charAt(i)) {
                same = false;
                break;
            }
        }
        if (!same) {
            return strs[0].substring(0, i);
        }
    }
    return strs[0].substring(0, minLength);
};